lib/datasource.js

```diff
@@ -2648,6 +2648,17 @@ DataSource.prototype.execute = function(command, args = [], options = {}) {
   }
 };
 
+/**
+ * Begin a new Transaction calling the underlying connector's begin method.
+ *
+ *
+ * @param [options] Options {isolationLevel: '...', timeout: 1000}
+ * @returns Promise A promise which resolves to a Transaction object
+ */
+DataSource.prototype.beginTransaction = async function(options) {
```

Comment:
What is the relation between the new `beginTransaction` function and the existing `DataSource.prototype.transaction`(https://github.com/strongloop/loopback-datasource-juggler/blob/36135df4c9b5575821a6922316b972c1cc23f6cc/lib/datasource.js#L2235)?

Reply:
The `beginTransaction` one just gives you a Transaction instance with a uuid you can use for your model instance to make certain db queries with whereas `transaction` gives you back transaction object that is tied to all the models attached to your datasource. IIUC we don't want a certain repository class which ties one model to a datasource to be able to use the higher level `transaction` API for this reason (shouldn't be accessing other models). See https://loopback.io/doc/en/lb3/Using-database-transactions.html#transaction-apis for more info.